src/test/lib/ddl/ddl_statement_test.cpp

```diff
@@ -0,0 +1,55 @@
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base_test.hpp"
+#include "tasks/chunk_compression_task.hpp"
+
+#include "SQLParser.h"
+#include "SQLParserResult.h"
+
+#include "types.hpp"
+#include "hyrise.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sql/sql_plan_cache.hpp"
+
+namespace opossum {
+
+class DDLStatementTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    Hyrise::reset();
+
+    // We reload table_a every time since it is modified during the test case.
+    _table_a = load_table("resources/test_data/tbl/int_float_create_index_test.tbl", 2);
+    Hyrise::get().storage_manager.add_table("table_a", _table_a);
+    auto compression_task_0 = std::make_shared<ChunkCompressionTask>("table_a", ChunkID{0});
+    auto compression_task_1 = std::make_shared<ChunkCompressionTask>("table_a", ChunkID{1});
+
+    Hyrise::get().scheduler()->schedule_and_wait_for_tasks({compression_task_0, compression_task_1});
+  }
+
+  // Tables modified during test case
+  std::shared_ptr<Table> _table_a;
+
+  const std::string _create_index = "CREATE INDEX myindex ON table_a (a)";
+};
+
+TEST_F(DDLStatementTest, CreateIndex) {
+  auto sql_pipeline = SQLPipelineBuilder{_create_index}.create_pipeline();
+
+  const auto& [pipeline_status, table] = sql_pipeline.get_result_table();
+  EXPECT_EQ(pipeline_status, SQLPipelineStatus::Success);
+
+  auto targeted_table = Hyrise::get().storage_manager.get_table("table_a");
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = targeted_table->indexes_statistics().at(0);
```

Comment:
Besides checking the index statistics, please also check the existence of the created index structures. Indexes are stored in a vector of `AbstractIndex` shared pointers in chunk objects (see the member variable `Chunk::_indexes`. `AbstractIndex::is_index_for` might be helpful to check if existing indexes are created for the correct segment(s).

Reply:
Thank you for the hint. `_indexes` is a private variable, so we built our check the other way round, which should be also sufficient.

Reply:
Verifying that the index statistics were created does not verify the successful creation of the index structures.

If private or protected members are to be tested, the test class can be declared as `friend class` (see https://en.cppreference.com/w/cpp/language/friend). An example can be seen in the `SettingsManager` where `SettingsManagerTest` is the friend class. Therefore, the protected functions `_add` and `_remove` can be used in `SettingsManagerTest`.

Besides the member variable `Chunk::_indexes`, `Chunk::get_indexes(...)` and `Chunk::get_index(...)` are public functions which can be used to retrieve the created index structures.

Reply:
The check I was referring to in my answer is not the table index statistics check. If you take a look at the last state of this class in the create index branch, we already use the get_index methods of the chunk. But the friends class is also a good hint, thank you.